src/DurableTask.AzureStorage/Partitioning/TablePartitionManager.cs

```diff
@@ -0,0 +1,795 @@
+            if(this.connectionString == null)
+            {
+                throw new Exception("Connection string is null. Managed identity is not supported in the table partition manager yet.");
+            }
+            this.options = new LeaseCollectionBalancerOptions
+            {
+                AcquireInterval = this.settings.LeaseAcquireInterval,
+                RenewInterval = this.settings.LeaseRenewInterval,
+                LeaseInterval = this.settings.LeaseInterval,
+                ShouldStealLeases = true
+            };
+            this.partitionManagerCancellationSource = new CancellationTokenSource();
+            this.partitionTable = new TableClient(this.connectionString, this.settings.PartitionTableName);
+            this.tableLeaseManager = new TableLeaseManager(this.partitionTable, this.service, this.settings, this.storageAccountName, this.options);
+        }
+
+        
+        /// <summary>
+        /// This method create a new instance of the class TableLeaseManager that represents the worker. 
+        /// And then start the loop that the worker keeps operating on the table. 
+        /// </summary>
+        async Task IPartitionManager.StartAsync()
+        {
+            await Task.Factory.StartNew(() => this.PartitionManagerLoop(this.partitionManagerCancellationSource.Token));
+            this.settings.Logger.PartitionManagerInfo(
+                this.storageAccountName,
+                this.settings.TaskHubName,
+                this.settings.WorkerId,
+                "", //Empty string as it does not target any particular partition, but rather only initiates the partition manager.
+                $"Starts acquiring and balancing leases.");
+        }
+
+        async Task PartitionManagerLoop(CancellationToken token)
+        {
+            while (!token.IsCancellationRequested)
+            {
+                TimeSpan timeToSleep = this.options.AcquireInterval;
+                try
+                {
+                    ReadTableReponse response = await this.tableLeaseManager.ReadAndWriteTable();
+                    if (response.WorkOnRelease || response.WaitForPartition)
+                    {
+                        timeToSleep = TimeSpan.FromSeconds(1);
+                    }
+                }
+                catch(RequestFailedException ex) when (ex.Status == 412)
+                {
+                    // if the worker failed to update the table, re-read the table immediately without waiting.
+                    timeToSleep = TimeSpan.FromSeconds(0);
+                }
+                catch(Exception exception)
+                {
+                    this.settings.Logger.PartitionManagerError(
+                       this.storageAccountName,
+                       this.settings.TaskHubName,
+                       this.settings.WorkerId,
+                       "",
+                       $"Errors in partition manager reading the table : {exception}");
```

Comment:
The message says "reading the table" but the failure could also be in writing the table, correct? I suggest we change the wording here to avoid confusion.
```suggestion
                       $"An unexpected failure occurred while trying to manage table partition leases: {exception}");
```